tests/coverage/coverage.cpp

```diff
@@ -114,20 +114,89 @@ TEST(Coverage, units)
     EXPECT_EQ(n, uc.name());
 }
 
-TEST(Coverage, when)
+TEST(Coverage, unitsGetVariations)
 {
-    const std::string id = "id";
+    libcellml::Model m;
+
+    libcellml::UnitsPtr u = std::make_shared<libcellml::Units>();
+    u->setName("a_unit");
 
-    libcellml::When w;
-    libcellml::When wm;
+    u->addUnit(libcellml::Units::StandardUnit::AMPERE, "micro");
+    m.addUnits(u);
 
-    w.setId(id);
+    libcellml::UnitsPtr un = m.units(0);
+    EXPECT_EQ("a_unit", un->name());
```

Comment:
I don't understand the variable names here ... what's the pattern I'm missing?

Reply:
This is part of the coverage tests so things get a little ugly here making sure we make use of copy constructors and things like that.  It has turned into the wild west of the tests.  Which is ok as they are not demonstrating any use just covering bits and pieces that haven't been dealt with in the 'use driven' tests.